src/utils.py

```diff
@@ -0,0 +1,406 @@
+        da_clip = da_clip.astype("uint8")
+        da_clip.rio.to_raster(save_dir / f"{filestem}_aoi{ext}", driver="COG")
+
+
+def clip_asap_inseason_trimester(start_month: int = 1):
+    """Clips existing global inseason trimester files to AOI"""
+    # Note: might crash. Adjust start_month to pick up where you left off.
+    aoi = load_codab_aoi()
+    load_dir = DATA_DIR / "public/processed/glb/asap/season/trimester_any_sen"
+    save_dir = DATA_DIR / "public/processed/sah/asap/season/trimester_any_sen"
+    for month in tqdm(range(start_month, 13)):
+        rel_months_str = "-".join(
+            [
+                str(x) if x < 13 else str(x - 12)
+                for x in range(month, month + 3)
+            ]
+        )
+        filestem = f"any_dekad_inseason_months-{rel_months_str}_sen"
+        ext = ".tif"
+        da = (
+            rxr.open_rasterio(load_dir / f"{filestem}{ext}")
+            .astype(float)
+            .squeeze(drop=True)
+        )
+        da.rio.write_crs(4326, inplace=True)
+        da_clip = da.rio.clip(aoi.geometry, all_touched=True)
+        da_clip = da_clip.fillna(254)
+        da_clip = da_clip.astype("uint8")
+        da_clip.rio.to_raster(save_dir / f"{filestem}_aoi{ext}", driver="COG")
+
+
+def clip_asap_raw():
+    """Clips ASAP start/end/senescence to AOI"""
+    codab = load_codab_all()
+    load_dir = DATA_DIR / "public/raw/glb/asap/reference_data"
+    save_dir = DATA_DIR / "public/processed/sah/asap"
+    for season in [1, 2]:
+        for s_e in ["s", "e", "sen"]:
+            filestem = f"pheno{s_e}{season}_v03"
+            ext = ".tif"
+            da = rxr.open_rasterio(load_dir / f"{filestem}{ext}").astype(
+                "uint8"
+            )
+            da = da.assign_attrs({"_FillValue": 254})
+            da = da.rio.clip(codab["geometry"], all_touched=True)
+            da = da.squeeze(drop=True)
+            da.rio.to_raster(save_dir / f"{filestem}_sah{ext}", driver="COG")
+
+
+def process_codab_all():
+    """
+    Combine CODABs for Burkina, Niger, and Chad.
+    Standardize column naming for admin names and codes.
+    Returns
+    -------
+
+    """
+    adm0s = [
+        {
+            "iso3": "bfa",
+            "name": "Burkina Faso",
+            "adm2": {"name_col": "ADM2_FR", "code_col": "ADM2_PCODE"},
+            "adm1": {"name_col": "ADM1_FR", "code_col": "ADM1_PCODE"},
+        },
+        {
+            "iso3": "ner",
+            "name": "Niger",
+            "adm2": {"name_col": "adm_02", "code_col": "rowcacode2"},
+            "adm1": {"name_col": "adm_01", "code_col": "rowcacode1"},
+        },
+        {
+            "iso3": "tcd",
+            "name": "Chad",
+            "adm2": {"name_col": "admin2Name", "code_col": "admin2Pcod"},
+            "adm1": {"name_col": "admin1Name", "code_col": "admin1Pcod"},
+        },
+    ]
+    gdfs = []
+    for adm0 in adm0s:
+        codab = CodAB(create_country_config(adm0.get("iso3")))
```

Comment:
Yeah I think when doing things on a more global FieldMaps is nice to not worry about column names and stuff.

My only concern is that it doesn't seem to be as up-to-date as the ones on HDX. For example in Mali, "Ménaka" used to be an admin2, but now it's an admin1 (as of 2016). The one on HDX has this, but FieldMaps still has the old arrangement. I noticed this for some other country too, can't recall which one now though. So for where it's feasible to use the HDX ones, I think I'm going to stick to those, since I assume that's what everyone else in OCHA would be using (or at least, should be).

Reply:
Basically for the global monitoring it might make more sense to use FieldMaps (or FAO Gaul, which I believe is also somewhat out of date), but for the Sahel, there's only three countries, so we can manage with the column name messiness

Reply:
Yeah, the base FieldMaps has some wonky stuff, but was noting because the FieldMaps CODs page I linked above is NOT the base FieldMaps, and each country scrapes updated data from HDX, so for instance the Mali CODs from FieldMaps has ML10 admin 1 area as Ménaka. You can still use the HDX cods of course, was just noting in case you weren't aware.